cartographer_ros/cartographer_ros/node_options.cc

```diff
@@ -24,6 +24,25 @@
 
 namespace cartographer_ros {
 
+std::array<double, 36> ReadCovariance(
+    ::cartographer::common::LuaParameterDictionary* const
+        lua_parameter_dictionary) {
+  std::array<double, 36> covariance = {};
```

Comment:
From a conceptual point of view, a pose covariance doesn't exist in Cartographer - a `PoseStamped` would be totally sufficient to describe the trajectory without TF. Adding constant parameters for the covariance to Cartographer's configuration therefore doesn't make sense, as these are specific to your type of filter pipeline and don't have any use in Cartographer.

But then, you can also just use a different TF topic name as @ojura suggested.

Reply:
> On the other hand I thought covariance is very related to the some cartographer settings, e.g. as voxel size.

Of course parameters like voxel size contribute to a systematic uncertainty. But defining a fixed pose covariance seems like a special requirement. And since it isn't used in Cartographer, it can be safely done outside of it. 

> So do you suggest to just use a different tf topic and obtaining the pose by subscribing to that topic?

I don't think this works since we also subscribe to TF (forget what I initially wrote). Maybe you can play around with the [`published_frame`](https://google-cartographer-ros.readthedocs.io/en/latest/configuration.html) parameter to publish to a child frame that doesn't affect the rest of the tree? (not quite sure at the moment if this will work). If not, I'd use a `PoseStamped`.

Reply:
> Of course parameters like voxel size contribute to a systematic uncertainty. But defining a fixed pose covariance seems like a special requirement. And since it isn't used in Cartographer, it can be safely done outside of it.

Agreed, I'll remove the covariance part.

> I don't think this works since we also subscribe to TF (forget what I initially wrote). 

In my understanding tf is not actually used by cartographer, but static transforms (base_link->imu, base_link->lidar) are read directly from an URDF.
However, cartographer_ros makes use of the tf tree to transform from the `tracked_frame` to the `published_frame`. Hence, if we publish the `tracked_frame` it works fine. when using a different tf topic.

> Maybe you can play around with the `published_frame` parameter to publish to a child frame that doesn't affect the rest of the tree? (not quite sure at the moment if this will work). If not, I'd use a PoseStamped.

I tried that, but unfortunately it always leads to having either two separate trees (causes visualization issues and warnings) or having wrong information in the tree.

So my suggestion:
I will introduce a flag to publish a `PoseStamped` of the `tracked_frame`, but will remove the flag for disabling tf publishing, since this issue can be solved by remapping tf. (Although I still think it's unreasonable to crowd the TCP/IP stack with data we are not using..)